Remotion/Web/Development.WebTesting/WebTestHelper.cs

```diff
@@ -190,12 +191,17 @@ public void OnSetUp ([NotNull] string testName)
     /// Creates a new browser session using the configured settings from <see cref="WebTestConfigurationFactory"/>.
     /// </summary>
     /// <param name="maximizeWindow">Specified whether the new browser session's window should be maximized.</param>
+    /// <param name="commandTimeout">Specifies a command timeout for the communication between the Selenium language bindings and the <see cref="IWebDriver"/>.</param>
     /// <returns>The new browser session.</returns>
-    public IBrowserSession CreateNewBrowserSession (bool maximizeWindow = true)
+    public IBrowserSession CreateNewBrowserSession (bool maximizeWindow = true, TimeSpan? commandTimeout = null)
     {
+      var testInfrastructureConfiguration = _testInfrastructureConfiguration;
+      if (commandTimeout != null)
+        testInfrastructureConfiguration = ((TestInfrastructureConfiguration) _testInfrastructureConfiguration).Clone (commandTimeout: commandTimeout);

```

Comment:
That cast is showing a design problem we have with the whole Clone-the-configuration idea.

From the list of not-so-practical options:
* Make a static factory method on the TestInfrastractureConfiguration type.
* Add all paramters to the clone method for overriding
* Come up with a different design for passing/overriding the command timeout.

For now, it's probably option 3.

Reply:
We have removed the Clone and added a new type `DriverOptions` which can be used to overwrite the command timeout from `TestInfrastructure`.

Pros of `DriverOptions`:

* The intent is conveyed very clearly by the type name, while passing a `TimeSpan? driverCommandTimeout` would feel a little misplaced at this location.
* It is easily extensible if future changes should require it.
* Future additions of properties to the class would not cause breaking changes, while substituting the class for a simple `TimeSpan? driverCommandTimeout` would.

Cons of `DriverOptions`:

* It has to be `[CanBeNull]` from its first appearance to its use, since overwriting the timeout set in `TestInfrastructureConfiguration` has to happen in in `IBrowserFactory`, where they meet.
* It contains only a single property and adding other properties currently makes no sense.